armory/scenarios/image_classification_load_adv_ds.py

```diff
@@ -0,0 +1,127 @@
+            train_data = load_dataset(
+                config["dataset"],
+                epochs=fit_kwargs["nb_epochs"],
+                split_type="train",
+                preprocessing_fn=preprocessing_fn,
+            )
+            if defense_type == "Trainer":
+                logger.info(f"Training with {defense_type} defense...")
+                defense = load_defense_wrapper(config["defense"], classifier)
+                defense.fit_generator(train_data, **fit_kwargs)
+            else:
+                classifier.fit_generator(train_data, **fit_kwargs)
+
+        if defense_type == "Transform":
+            # NOTE: Transform currently not supported
+            logger.info(f"Transforming classifier with {defense_type} defense...")
+            defense = load_defense_wrapper(config["defense"], classifier)
+            classifier = defense()
+
+        classifier.set_learning_phase(False)
+
+        # Evaluate the ART classifier on benign test examples
+        logger.info(f"Loading test dataset {config['dataset']['name']}...")
+        test_data_generator = load_dataset(
+            config["dataset"],
+            epochs=1,
+            split_type="test",
+            preprocessing_fn=preprocessing_fn,
+        )
+        logger.info("Running inference on benign examples...")
+        metrics_logger = metrics.MetricsLogger.from_config(config["metric"])
+
+        for x, y in tqdm(test_data_generator, desc="Benign"):
+            y_pred = classifier.predict(x)
+            metrics_logger.update_task(y, y_pred)
+        metrics_logger.log_task()
+
+        # Evaluate the ART classifier on adversarial test examples
+        logger.info("Generating / testing adversarial examples...")
+
+        ###################################################
+        # test_data_generator = load_dataset(
+        #     config["dataset"],
+        #     epochs=1,
+        #     split_type="test",
+        #     preprocessing_fn=preprocessing_fn,
+        # )
+
+        # for x, y in tqdm(test_data_generator, desc="Attack"):
+        #     x_adv = attack.generate(x=x)
+        #     y_pred_adv = classifier.predict(x_adv)
+        #     metrics_logger.update_task(y, y_pred_adv, adversarial=True)
+        #     metrics_logger.update_perturbation(x, x_adv)
+        ####################################################
+        attack = load_attack(config["attack"], classifier)
+
+        attack_type = config["attack"]["type"]
+        if attack_type == "transfer":
+            # test_data_generator = attack     # suggested by DSlater but load_attack loads an ART attack function, not the attack dataset, right now. Plus, the dataset in the config json file is the adversarial dataset anyways, so load_dataset makes sense
+            test_data_generator = load_dataset(
+                config["dataset"],
+                epochs=1,
+                split_type="adv",
+                preprocessing_fn=preprocessing_fn,
+            )
```

Comment:
This, if I understand correctly, would do attacks on the same images as those trained on above. This should be split into train/test.